.github/workflows/ci.yml

```diff
@@ -57,4 +57,39 @@ jobs:
           cache: gradle
 
       - name: Compile Java Example app
-        run: ./gradlew :example:assemble
\ No newline at end of file
+        run: ./gradlew :example:assemble
+
+  checkApiCompatibility:
+    runs-on: ubuntu-latest
+    # Prevent duplicate builds on internal PRs.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Generate current API signatures
+        run: ./gradlew :centrifuge:metalavaGenerateTempSignature
+
+      - name: Calculate API diff
+        run: |
+          set +e
+          git diff --no-index centrifuge/api.txt centrifuge/build/metalava/current.txt
+          set -e
+
+# TODO: uncomment this when permisison to comment on PR is available
+#      - uses: int128/diff-action@v1
+#        with:
+#          token: ${{ secrets.GITHUB_TOKEN }}
+#          base: centrifuge/api.txt
+#          head: centrifuge/build/metalava/current.txt
+#          label: api-changed
```

Comment:
This should post comment to PR with diff in API Could be nice to see it in PRs, but not really essential.
The job doesn't have permission to post a comment and the step fails.

It's either because of repository settings (GitHub Actions don't have WRITE permissions) or something else.
One way to solve it is to create secret with PAT that has `repo: read` and `pull_requests: write` permission.

Commented out for now...